apis/python/src/tiledbsoma/dense_nd_array.py

```diff
@@ -152,7 +153,7 @@ def read_tensor(
             result_order, accept=["column-major", "row-major"]
         )
         with self._tiledb_open("r") as A:
-            target_shape = _dense_index_to_shape(coords, A.shape, result_order)
+            target_shape = dense_indices_to_shape(coords, A.shape, result_order)
```

Comment:
there is a bug here - `coords` param can contain a pa.Array, which is not handled by this helper.

The actual bug is the definition of `DenseNdCoordinates`, which should not accept a pa.Array (a bug I left behind when I did the sub-array changes - sorry!)   the correct fix IMHO is to change the definition of `DenseNdCoordinates` to be only a list of int | slice

Reply:
@bkmartinjr I fixed this on #574: I'll backport

Reply:
gotcha - I'm OK either way as long as it gets fixed at some point ;)  Do whichever is easiest for you!